src/components/DiagramFrame/FramePanels/DiagramSettings.tsx

```diff
@@ -82,46 +81,45 @@ export const DiagramSettings: React.FC<DiagramSettingsProps> = ({
         {modelName && (
-          <div>
-            <SpaceVertical size="xxsmall">
-              <FieldSelect
-                options={branchOpts ? branchOpts : []}
-                placeholder="Loading Git Branches..."
-                label="Current Branch"
-                value={gitBranch && gitBranch.name}
-                onChange={(value)=>{setSelectedBranch(value)}}
-                disabled={(gitBranch && gitBranch.is_production) || !diagramExplore}
-              />
-              <Label>Select an Explore</Label>
-              <ExploreList>
-                {exploreList && exploreList.map((explore: ExploreDropdown, index: number) => {
-                  return (
-                    <ExploreListitem key={`explore-${index}`} style={{backgroundColor: getExploreListItemBackgroundColor(explore.value, currentExplore, diagramExplore)}}>
-                      <ExploreButton
-                        onClick={() => {
-                          selectionInfo.lookmlElement === "explore" || setSelectionInfo({})
-                          setViewVisible({})
-                          setZoomFactor(ZOOM_INIT)
-                          setViewPosition({x: X_INIT, y: Y_INIT})
-                          setMinimapUntoggled(true)
-                          setMinimapEnabled(false)
-                          history.push(
-                            internalExploreURL({
-                              model: currentModel.name,
-                              explore: explore.value
-                            })
-                          )
-                        }}
-                        value={explore.value}
-                      >
-                        {explore.label}
-                      </ExploreButton>
-                    </ExploreListitem>
-                  )
-                })}
-              </ExploreList>
-            </SpaceVertical>
-          </div>
+          <>
+            <FieldSelect
+              options={branchOpts ? branchOpts : []}
+              placeholder="Loading Git Branches..."
+              label="Current Branch"
+              value={gitBranch && gitBranch.name}
+              onChange={(value)=>{setSelectedBranch(value)}}
+              disabled={(gitBranch && gitBranch.is_production) || !diagramExplore}
+            />
+            <Divider appearance="light" my="medium" />
+            <Label fontSize="small" style={{marginTop: "0rem"}}>Select an Explore</Label>
+            <ExploreList style={{marginTop: theme.sizes.xxxsmall, overflow: "auto"}}>
+              {exploreList && exploreList.map((explore: ExploreDropdown, index: number) => {
+                return (
+                  <ExploreListitem key={`explore-${index}`} style={{backgroundColor: getExploreListItemBackgroundColor(explore.value, currentExplore, diagramExplore)}}>
+                    <ExploreButton
+                      onClick={() => {
+                        selectionInfo.lookmlElement === "explore" || setSelectionInfo({})
```

Comment:
consider externalizing this function from the JSX. Can this functionality be moved to the ExploreButton itself?